interface/wx/maskededit.h

```diff
@@ -0,0 +1,935 @@
+
+        @param alignment
+            @c wxALIGN_LEFT (default) or @c wxALIGN_RIGHT.
+
+    */
+    wxEditFieldFlags& SetAlignment(wxAlignment alignment);
+
+    /**
+        Sets the fill char of this wxEditFieldFlags to @a fillchar.
+
+        The fill char is used to fill the empty (i.e. not typed yet)
+        positions in the field for showing them. Typically it is a space.
+
+        @param fillchar
+            Any printable character.
+
+    */
+    wxEditFieldFlags& SetFillChar(wxChar fillchar);
+
+    /**
+        Sets the padding char of this wxEditFieldFlags to @a paddingchar.
+
+        The padding char is used to fill the empty (i.e. not typed yet)
+        positions in the field for the return of wxMaskedEdit::GetPlainValue()
+        and wxMaskedEdit::GetFieldValue().
+
+        Default is '\0' (null) which is interpreted as "don't fill"
+
+        @param paddingchar
+            Any printable character or '\0'.
+
+    */
+    wxEditFieldFlags& SetPaddingChar(wxChar paddingchar);
+
+    /**
+        Returns the alignment used in this object.
+
+        @see SetAlignment()
+    */
+    wxAlignment GetAlignment();
+
+    /**
+        Returns the fill char used in this object.
+
+        @see SetFillChar()
+    */
+    wxChar GetFillChar();
+
+    /**
+        Returns the padding char used in this object.
+
+        @see SetPaddingChar()
+    */
+    wxChar GetPaddingChar();
+};
+
+/** @addtogroup group_maskededit_predef
+/**@{*/
+/**
+    The prototype of function used to test a field of a masked edit control.
+
+    The function accepts as parameters a pointer to the instance of the object
+    that calls this function (your wxMaskedEditText or wxMaskedEditCombo), the
+    field for which this function applies and a pointer to an object containing
+    needed information for this function.
+
+    An example of a currently defined function is wxMaskedRangeCheck().
+
+    @library{wxcore}
+    @category{ctrl}
+
+    @see wxMaskedEdit::SetFieldFunction()
+
+    @since 3.1.6
+
+*/
+typedef long wxMaskedFieldFunc(const wxMaskedEdit*, size_t, void*);
+
+/**
+    The prototype of function used to test the value of a masked edit control.
+
+    The function accepts as parameters a pointer to the instance of the object
+    that calls this function (your wxMaskedEditText or wxMaskedEditCombo) and
+    a pointer to an object containing needed information for this function.
+
+    An example of a currently defined function is wxMaskedDateShort().
+
+    @library{wxcore}
+    @category{ctrl}
+
+    @see wxMaskedEdit::SetControlFunction()
+
+    @since 3.1.6
+
+    @header{wx/maskededit.h}
+*/
+typedef long wxMaskedFunc(const wxMaskedEdit*, void*);
+/**@}*/
+
+/**
+    @class wxMaskedEditParams
+
+    Container used by wxMaskedEdit to store all configurable parameters.
+
+    It is also useful if you have several masked edit controls and you want
+    to set all parameters at once repeatedly.
```

Comment:
I think this should be made private, the explanation about why it is useful doesn't seem very convincing, it would be just as simple to create an initialization function and call it with different controls.

I.e. I really don't understand why should we have this for this control, when we don't have anything equivalent for any other one -- how is it different?

Reply:
This is also done so I'm marking it as "Resolved".